File: backend/baitaplon/src/main/java/com/gianglt/baitaplon/model/Payment.java
package com.gianglt.baitaplon.model;

import com.gianglt.baitaplon.domain.PaymentMethod;
import com.gianglt.baitaplon.domain.PaymentStatus;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.math.BigDecimal;
import java.time.Instant;

@Entity
@Table(name = "payments")
@Getter @Setter
@NoArgsConstructor @AllArgsConstructor @Builder
public class Payment {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "order_id", nullable = false, unique = true)
    private Order order;

    @Enumerated(EnumType.STRING)
    private PaymentMethod method;

    @Enumerated(EnumType.STRING)
    private PaymentStatus status;

    private String provider;
    private BigDecimal amount;
    private String txnRef;

    private String bankCode;
    private String bankTranNo;
    private String transactionNo;
    private String responseCode;
    private String secureHash;
    private String orderInfo;
    private String payDate;

    @Column(length = 4000)
    private String rawReturnQuery;

    @CreationTimestamp
    private Instant createdAt;
}

File: frontend/angular-ecommerce-main/src/app/shared/models/payment.ts
// src/app/shared/models/payment.ts
import { CartProduct } from './cart-product';

export interface PaymentDto {
  products: CartProduct[];
  total: number;
}

export type PaymentMethod =
  | 'VNPAY' | 'COD' | 'MOMO' | 'BANK' | 'OTHER';

export type PaymentStatus =
  | 'PENDING' | 'PAID' | 'FAILED' | 'CANCELLED' | 'REFUNDED';

export interface Payment {
  id: number;

  // BE trả orderId (Long) -> FE dùng số, không phải object lồng order:{id}
  orderId?: number | null;

  method?: PaymentMethod;
  status?: PaymentStatus;
  provider?: string | null;
  amount?: number;
  txnRef?: string | null;

  bankCode?: string | null;
  bankTranNo?: string | null;
  transactionNo?: string | null;
  responseCode?: string | null;

  // BE có thêm 2 field này, thêm để hiển thị nếu cần
  orderInfo?: string | null;
  payDate?: string | null;

  createdAt?: string | null;
}
